File: backend/app/schemas/teams.py
import datetime

from pydantic import BaseModel

class Last5GameInfoOut(BaseModel):
    """Output model for a team's last 5 games info"""
    game_id: int
    date: int
    home_team_tri_code: str
    away_team_tri_code: str
    home_score: int | None
    away_score: int | None

    class Config:
        from_attributes = True

class TeamBasicInfoOut(BaseModel):
    """Output model for a team's basic info"""
    name: str
    tricode: str
    logoUrl: str | None

class TeamScheduledGameInfoOut(BaseModel):
    """Output model for a team's upcoming game info"""
    id: int
    date: str
    homeTeam: TeamBasicInfoOut
    awayTeam: TeamBasicInfoOut
    awayScore: int | None
    homeScore: int | None
    time: datetime.datetime
    venue: str
    gameState: str
    predictions: TeamGamePredictionOut | None
    moneyline: TeamMoneylineOut | None
    isNextGame: bool

class TeamRosteredPlayer(BaseModel):
    """Output model for a player on a team's roster"""
    id: int
    headshot: str
    first_name: str
    current_team_tri_code: str
    position: str
    last_name: str
    number: int | None
    shoots_catches: str

class TeamSearchResultOut(BaseModel):
    """Output model for a team search result"""
    name: str
    tricode: str
    logoUrl: str | None

class TeamSidePrediction(BaseModel):
    """Prediction details for one side (home or away) of a game"""
    tri_code: str
    prob_win: float

class TeamGamePredictionOut(BaseModel):
    """Output model for game-level predictions (both teams)"""
    home: TeamSidePrediction
    away: TeamSidePrediction

class TeamMoneylineOut(BaseModel):
    """Output model for game-level moneyline predictions (both teams)"""
    home: int
    away: int

File: vite-frontend/src/types/teams.ts
export interface Team {
  tricode: string;
  name: string;
  logoUrl: string;
}

export interface PredictedScore {
  away: number;
  home: number;
}

export interface TeamScheduledGame {
  id: number;
  date: string;
  time: string;
  venue: string;
  awayTeam: Team;
  homeTeam: Team;
  awayScore: number | null;
  homeScore: number | null;
  gameState: string | null;
  predictions?: TeamGamePrediction;
  moneyline?: TeamMoneyline;
  isNextGame?: boolean;
}

export interface SearchTeamResult {
  name: string;
  tricode: string;
  logoUrl: string | null;
}

export interface TeamPredictionSide {
  tri_code: string;
  prob_win: number;
}

export interface TeamGamePrediction {
  home: TeamPredictionSide;
  away: TeamPredictionSide;
}

export interface TeamMoneyline {
  home: number;
  away: number;
}